warden/client.go

```diff
@@ -0,0 +1,93 @@
+package warden
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/goto/dex/pkg/errors"
+)
+
+type Client struct {
+	BaseURL string
+	Client  *http.Client
+}
+
+func NewClient(baseURL string) *Client {
+	return &Client{
+		BaseURL: baseURL,
+		Client:  &http.Client{},
+	}
+}
+
+func (c *Client) ListUserTeams(ctx context.Context, req TeamListRequest) ([]Team, error) {
+	const (
+		endpoint      = "/api/v1"
+		userPath      = "/users/"
+		teamsEndpoint = "/teams"
+	)
+	url := fmt.Sprintf("%s%s%s%s%s", c.BaseURL, endpoint, userPath, req.Email, teamsEndpoint)
```

Comment:
we dont need to create const for each of the url path, we can just

```suggestion
	url := fmt.Sprintf("%s/api/v1/users/%s/teams", c.BaseURL, req.Email)
```